### DISCUSSION TITLE
draw function

### DISCUSSION THREAD

**Discussion by idoglanz** [IfcOpenShell/IfcOpenShell#3535] [Q&A] [answered]
Hi! I'm a bit new to this library but so far its been great.

I'm trying to export an SVG of each floor of an IFC file, what would be the best approach? Split the IFC by floor and save each? (tried that but the ifcpatch Split for some reason generates an IFC file which I can't later export to svg -> no geometries detected...). Could I set the Z axis of the draw function so it will only show a specific floor (assuming i load the whole model)?

And in general are there anywhere some examples of draw.py usage? 

Thank you!

**Comment by Moult** [comment]
Unfortunately I'm not aware of any documentation. Does auto_floorplan do what you're after?

**Reply by idoglanz** [reply]
Thanks for the quick response.

Unfortunately auto_floorplan just gives me a top view of the entire model:
<img width="1171" alt="image" src="https://github.com/IfcOpenShell/IfcOpenShell/assets/44923186/802d1746-d693-4b6c-a9e7-02eadcf88fbd">


This file I'm working with is -
[test_2.ifc.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/12238541/test_2.ifc.txt)

And the code (which I might be using wrong) is - 
```
import ifcopenshell
import ifcopenshell.draw

filename = "./data/ifc_files/test_2.ifc"
ifc_file = ifcopenshell.open("./data/ifc_files/test_2.ifc")

output_file = filename.replace(".ifc", ".svg")
draw_settings = ifcopenshell.draw.draw_settings(auto_floorplan=True,)
output_data = ifcopenshell.draw.main(draw_settings, files=[ifc_file,])

open(output_file, "wb").write(output_data)
```

Thanks!

**Reply by Moult** [reply]
Are you sure that's the right file? That looks like a totally different building.

**Reply by idoglanz** [reply]
Sorry thats the wrong one (same filename, different folder, my bad)

Heres the right one - 

[test_2.ifc.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/12238675/test_2.ifc.txt)

**Reply by Moult** [reply, marked as answer]
I tested and it seems to work for me (I turned of merge projection on my end since I have some issues on my local dev environment):

```
import ifcopenshell
import ifcopenshell.draw

filename = "/home/dion/tmp/test_2.ifc"
ifc_file = ifcopenshell.open(filename)

output_file = filename.replace(".ifc", ".svg")
draw_settings = ifcopenshell.draw.draw_settings(auto_floorplan=True,)
output_data = ifcopenshell.draw.main(draw_settings, files=[ifc_file,], merge_projection=False)

open(output_file, "wb").write(output_data)
```

I get an SVG which has all three floors superimposed one on top of another. If I go into Inkscape I can move them separately:

![2023-08-04-085817_1579x552_scrot](https://github.com/IfcOpenShell/IfcOpenShell/assets/88302/613e7b12-24d8-4b1c-a94e-52d88b649e52)

**Reply by idoglanz** [reply]
Oh nice thank you!
I hadn't imagined it would be one on top of the other, but indeed I can open it and move them around

Thanks man
